File: Lista-15/java/9.java
interface Imprimivel {
    String imprimir();
}

class Relatorio implements Imprimivel {
    @Override
    public String imprimir() {
        return "Imprimindo relatório...";
    }
}

class Contrato implements Imprimivel {
    @Override
    public String imprimir() {
        return "Imprimindo contrato...";
    }
}

public class Main {
    public static void main(String[] args) {
        Imprimivel[] documentos = {new Relatorio(), new Contrato()};
        for (Imprimivel doc : documentos) {
            System.out.println(doc.imprimir());
        }
    }
}

File: Lista-15/python/9.py
from abc import ABC, abstractmethod

class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self):
        pass

class Relatorio(Imprimivel):
    def imprimir(self):
        return "Imprimindo relatório..."

class Contrato(Imprimivel):
    def imprimir(self):
        return "Imprimindo contrato..."

# Exemplo de uso
documentos = [Relatorio(), Contrato()]
for doc in documentos:
    print(doc.imprimir())
